R/clean_names.R

```diff
@@ -40,24 +40,35 @@
 #' # library(readxl)
 #' # read_excel("messy_excel_file.xlsx") %>% clean_names()
 
-clean_names <- function(dat, case = c(
-                        "snake", "lower_camel", "upper_camel", "screaming_snake",
-                        "lower_upper", "upper_lower", "all_caps", "small_camel",
-                        "big_camel", "old_janitor", "parsed", "mixed"
-                      )) {
-  if(!is.data.frame(dat)){stop( "clean_names() must be called on a data.frame.  Consider janitor::make_clean_names() for other cases of manipulating vectors of names.") }
-  stats::setNames(dat, make_clean_names(names(dat), case = case))
+# create new clean_names method
+clean_names <- function(dat, case) {
```

Comment:
@sfirke Hey, I finally got around to making some changes. In order to make an `sf` method, I had to make a default method for `clean_names()` (which still only accepts a`data.frame`), so I could then make a sf method.